### PR TITLE
EC-350 Send comment for review

### PR DESCRIPTION
**Pull request by suegarner**
https://nicedigital.atlassian.net/browse/EC-350

### CODE REVIEW

File: Comments/ClientApp/src/components/SubmitResponseFeedback/SubmitResponseFeedback.js [hunk 1]
```diff
@@ -11,30 +11,43 @@ export const SubmitResponseFeedback = (props) => {
 		hasTobaccoLinks,
 		showExpressionOfInterestSubmissionQuestion,
 		organisationExpressionOfInterest,
+		emailAddress,
+		isOrganisationCommenter,
 	} = props;
 
 	let items = [];
+	const emailRegex = /\S+@\S+\.\S+/;
```
Comment (John-Williams) [inline Comments/ClientApp/src/components/SubmitResponseFeedback/SubmitResponseFeedback.js:19]: is this really necessary here too?

**Review by John-Williams — CHANGES_REQUESTED** [review]
i've added a few comments / questions. as always, they're up for debate.

thanks.

File: Comments/ClientApp/src/components/Review/Review.js [hunk 2]
```diff
@@ -341,6 +344,38 @@ export class Review extends Component<PropsType, StateType> {
...
+			}
+		});
+
+		load("submitToLead", undefined, [], {}, "POST", {
+			emailAddress,
+			comments,
+			answers: answersToSubmit,
+		}, true)
+			.then(() => {
+				this.setState({
```
[the remark this answers is not in the document]
Comment (johnholland-nice) [reply Comments/ClientApp/src/components/Review/Review.js:365]: Yes; I wasn't sure whether they would still be applicable or if we would need more. I will investigate.

File: Comments/Services/SubmitService.cs [hunk 3]
```diff
@@ -64,6 +66,57 @@ public SubmitService(ConsultationsContext context, IUserService userService, ICo
...
+			}
+			if (submission.Answers.Count > 0)
+			{
+				UpdateAnswersModel(submission.Answers, submissionToSave, submittedStatus);
+				var answerIds = submission.Answers.Select(a => a.AnswerId).ToList();
+				_context.DuplicateAnswer(answerIds);
+			}
+
+			//now for analytics calculate the number of seconds between the user's first comment or answer and the submission date
+			var earliestDate = submissionToSave.SubmissionComment.Any() ? submissionToSave.SubmissionComment.Min(sc => sc.Comment.CreatedDate) : DateTime.MaxValue;
```
Comment (John-Williams) [inline Comments/Services/SubmitService.cs:109]: is there any chance this block of code could be shared between the submissions please?

It's just quite a chunk of code that looks identical to me.

so it'd be something like:

Submission.DurationBetweenFirstCommentOrAnswerSavedAndSubmissionInSeconds = GetDuration(submissionToSave);

then all that code could be moved into a GetDuration function.

**Review by John-Williams — CHANGES_REQUESTED** [review]

**Review by John-Williams — APPROVED** [review]
thanks both!

**Pull request merged**